collectors/metrics/cmd/metrics-collector/main.go

```diff
@@ -282,17 +287,30 @@ type Options struct {
 }
 
 func (o *Options) Run() error {
-
 	var g run.Group
 
+	metricsReg := prometheus.NewRegistry()
+	metricsReg.MustRegister(
+		version.NewCollector("metrics-collector"),
+		collectors.NewGoCollector(
+			collectors.WithGoCollectorRuntimeMetrics(collectors.GoRuntimeMetricsRule{Matcher: regexp.MustCompile("/.*")}),
+		),
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+	)
+
+	// Some packages still use default Register. Replace to have those metrics.
+	prometheus.DefaultRegisterer = metricsReg
+
 	err, cfg := initConfig(o)
 	if err != nil {
 		return err
 	}
 
+	cfg.Registry = metricsReg
+
 	worker, err := forwarder.New(*cfg)
 	if err != nil {
-		return fmt.Errorf("failed to configure metrics collector: %v", err)
+		return errors.Wrap(err, "failed to configure metrics collector")
```

Comment:
any particular reason we would choose to import this package to handle errors in favour of stdlib?

Reply:
It's what we use for thanos upstream (or aiming to, promql-engine already uses it). This returns the stacktrace which is super useful at times. github.com/pkg/errors did similar I think, but is no longer maintained.

Reply:
I also prefer using the stdlib with `fmt.Errorf("%w")`. Regarding the stack trace, I don't see why we would need it. I mean, if so, it is a symptom of bad error handling that is not descriptive enough in the stack.
I don't like depending on those small libs while the stdlib does the job. But it's me 😉